File: backend/app/schemas/prescription.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional

class PrescriptionPageResponse(BaseModel):
    id: int
    file_url: str
    file_type: str
    page_number: int
    original_filename: Optional[str]
    file_size_bytes: Optional[int]

    class Config:
        from_attributes = True

class PrescriptionMedicineResponse(BaseModel):
    id: int
    extracted_name: str
    strength: Optional[str] = None
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    duration: Optional[str] = None
    instructions: Optional[str] = None
    confidence_score: Optional[float] = None
    needs_verification: bool

    class Config:
        from_attributes = True

class TestResponse(BaseModel):
    id: int
    test_name: str
    description: Optional[str] = None
    confidence_score: Optional[float] = None
    needs_verification: bool

    class Config:
        from_attributes = True

class PrescriptionResponse(BaseModel):
    id: int
    user_id: int
    status: str
    failure_reason: Optional[str] = None
    confidence_score: Optional[float] = None
    needs_verification: bool = False
    uploaded_at: datetime
    processed_at: Optional[datetime]
    pages: List[PrescriptionPageResponse]
    medicines: List[PrescriptionMedicineResponse] = []
    tests: List[TestResponse] = []

    class Config:
        from_attributes = True

class PrescriptionListResponse(BaseModel):
    id: int
    status: str
    uploaded_at: datetime
    page_count: int

    class Config:
        from_attributes = True

File: frontend/src/types/prescription.ts
export interface PrescriptionPage {
  id: number;
  file_url: string;
  file_type: string;
  page_number: number;
  original_filename: string | null;
  file_size_bytes: number | null;
}

export interface PrescriptionMedicine {
  id: number;
  extracted_name: string;
  original_extracted_name: string | null;
  normalized_name: string | null;
  suggested_matches: string | null; // JSON string array
  strength: string | null;
  dosage: string | null;
  frequency: string | null;
  duration: string | null;
  instructions: string | null;
  confidence_score: number | null;
  needs_verification: boolean;
  verified_by: number | null;
  verified_at: string | null;
}

export interface PrescriptionTest {
  id: number;
  test_name: string;
  description: string | null;
  confidence_score: number | null;
  needs_verification: boolean;
}

export interface Prescription {
  id: number;
  user_id: number;
  status: "uploaded" | "processing" | "completed" | "failed";
  failure_reason: string | null;
  confidence_score: number | null;
  needs_verification: boolean;
  uploaded_at: string;
  processed_at: string | null;
  pages: PrescriptionPage[];
  medicines: PrescriptionMedicine[];
  tests: PrescriptionTest[];
}

export interface PrescriptionListResponse {
  id: number;
  status: "uploaded" | "processing" | "completed" | "failed";
  uploaded_at: string;
  page_count: number;
  test_count?: number;
  medicines_summary?: string;
}
